packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx

```diff
@@ -0,0 +1,176 @@
+import { Modal, ModalHeader, ModalTransactionFooter } from '@/common/components/Modal'
+import {
+  /*StepDescriptionColumn, */ Stepper,
+  StepperModalBody,
+  StepperModalWrapper,
+} from '@/common/components/StepperModal'
+import { TextMedium } from '@/common/components/typography'
+import { useMachine } from '@/common/hooks/useMachine'
+import { useModal } from '@/common/hooks/useModal'
+import { SignTransactionModal } from '@/common/modals/SignTransactionModal/SignTransactionModal'
+import { isLastStepActive } from '@/common/modals/utils'
+import { getSteps } from '@/common/model/machines/getSteps'
+import { useYupValidationResolver } from '@/common/utils/validation'
+import { machineStateConverter } from '@/council/modals/AnnounceCandidacy/helpers'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+import { StyledStepperBody } from '@/proposals/modals/AddNewProposal'
+import { GroupIdToGroupParam } from '@/working-groups/constants'
+
+import { SuccessModal, CreateOpeningSteps as Steps, ImportOpening } from './components'
+import { createOpeningMachine, CreateOpeningMachineState, getTxParams } from './machine'
+import { OpeningConditions, CreateOpeningForm, CreateOpeningModalCall, OpeningSchema, defaultValues } from './types'
+
+export const CreateOpeningModal = () => {
+  const [showImport, setShowImport] = useState<boolean>(false)
+
+  const { api, connectionState } = useApi()
+  const { active: activeMember } = useMyMemberships()
+  const [state, send, service] = useMachine(createOpeningMachine)
+  const { hideModal, modalData } = useModal<CreateOpeningModalCall>()
+
+  const { group } = modalData
+  const workingGroupConsts = api?.consts[group]
+
+  const context = {
+    hiringTarget: 1,
+    minStake: new BN(50000),
+    group,
+    minUnstakingPeriodLimit: workingGroupConsts?.minUnstakingPeriodLimit,
+    minimumApplicationStake: workingGroupConsts?.minimumApplicationStake,
+  } as OpeningConditions
+  const path = useMemo(() => machineStateConverter(state.value), [state.value])
+  const resolver = useYupValidationResolver<CreateOpeningForm>(OpeningSchema, path)
+  const form = useForm<CreateOpeningForm>({ resolver, mode: 'onChange', defaultValues, context })
+  useEffect(() => {
+    form.trigger(machineStateConverter(state.value) as keyof CreateOpeningForm)
+  }, [machineStateConverter(state.value)])
+
+  const createOpeningTx = useMemo(() => {
+    if (api && group) {
+      const { ...specifics } = form.getValues() as CreateOpeningForm
+      const { description, stakePolicy, rewardPerBlock } = getTxParams(group, specifics)
+      return api.tx[group].addOpening(description, 'Regular', stakePolicy, String(rewardPerBlock))
+    }
+  }, [connectionState, activeMember?.id, form.formState.isValidating, group])
+
+  const { feeInfo } = useTransactionFee(activeMember?.controllerAccount, () => createOpeningTx, [
+    connectionState,
+    createOpeningTx,
+  ])
```

Comment:
You can use the shorter syntax here:
```suggestion
  const { transaction, feeInfo } = useTransactionFee(activeMember?.controllerAccount, () => {
    if (api && group) {
      const { ...specifics } = form.getValues() as CreateOpeningForm
      const { description, stakePolicy, rewardPerBlock } = getTxParams(group, specifics)
      return api.tx[group].addOpening(description, 'Regular', stakePolicy, String(rewardPerBlock))
    }
  }, [api?.isConnected, activeMember?.id, group, form.formState.isValidating])
```
(`transaction` is `createOpeningTx`)